### PR TITLE
feat: support authentication in tests

### PR DESCRIPTION
**Pull request by ojongerius**
Addresses #81 not super polished, but it gets the job done for now. I'm happy to refine as #58 is not in yet anyway 😄 

This currently fails against staging, as #58 has not been merged yet:

```sh
▶ yarn jest
yarn run v1.6.0
$ /Users/ojongerius/repos/fcc-open-api/node_modules/.bin/jest
 FAIL  test/user.test.js
  ✕ should return null users and authentication error without a token (35ms)
  ✕ should return null users and authentication error when using an invalid token (4ms)
  ✓ should return a user after one has been created (31ms)

  ● should return null users and authentication error without a token

    expect(received).toBe(expected) // Object.is equality

    Expected value to be:
      null
    Received:
      []

    Difference:

      Comparing two different types of values. Expected null but received array.

      32 |     graphqlSchema, query, rootValue, global.mockedContextWithOutToken);
      33 |
    > 34 |   expect(result.data.users).toBe(null);
      35 |   expect(result.errors).toHaveLength(1);
      36 |   expect(result.errors[0].message).toBe('You must supply a JSON Web Token for authorization!');
      37 | });

      at Object.<anonymous>.it (test/user.test.js:34:29)

  ● should return null users and authentication error when using an invalid token

    expect(received).toBe(expected) // Object.is equality

    Expected value to be:
      null
    Received:
      []

    Difference:

      Comparing two different types of values. Expected null but received array.

      43 |     graphqlSchema, query, rootValue, global.mockedContextWithInValidToken);
      44 |
    > 45 |   expect(result.data.users).toBe(null);
      46 |   expect(result.errors).toHaveLength(1);
      47 |   expect(result.errors[0].message).toBe('You are not authorized, jwt malformed');
      48 | });

      at Object.<anonymous>.it (test/user.test.js:45:29)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 1 passed, 3 total
Snapshots:   0 total
Time:        1.358s, estimated 3s
Ran all test suites.
error Command failed with exit code 1.
```

But passes against `Bouncey:feat/auth` merged with staging 🎈 

- [x] Fix availability of JWT_CERT.

### CODE REVIEW

File: test/user.test.js [hunk 1]
```diff
@@ -6,16 +6,55 @@ const UserModel = require('../dataLayer/model/user.js');
...
+    graphqlSchema,
+    query,
+    rootValue,
+    global.mockedContextWithOutToken
+  );
+
+  expect(result.data.users).toBe(null);
+  expect(result.errors).toHaveLength(1);
+  expect(result.errors[0].message).toBe(
+    'You must supply a JSON Web Token for authorization!'
```
Comment (ojongerius) [inline test/user.test.js]: Testing without a token and without a token is probably not what we want for each query and mutation. I'm happy to refactor that out into specific auth tests while reviewing this PR alternatively we wait until we have more queries to test.

File: test/user.test.js [hunk 2]
```diff
@@ -6,16 +6,55 @@ const UserModel = require('../dataLayer/model/user.js');
...
   await mongoose.connect(global.__MONGO_URI__);
 });
 
-/* eslint-disable no-undef */
 afterAll(async () => {
-  /* eslint-enable no-undef */
   await mongoose.disconnect();
 });
 
+// language=GraphQL
```
Comment (ojongerius) [inline test/user.test.js:16]: I can see having more helpers to create queries when we have more tests.

File: test/user.test.js [hunk 3]
```diff
@@ -6,16 +6,62 @@ const UserModel = require('../dataLayer/model/user.js');
...
+            name
+            email
+        }
+    }
+  `;
+
+const rootValue = {};
+
+it(
+  'should return null users and authentication error ' + ' without a token',
```
Comment (ojongerius) [inline test/user.test.js:29]: This is hilarious: if I remove splitting of the string, ie: `should return null users and authentication error 
 without a token` running of `yarn format` will have the following result:

'prettier' will pull line 29 and 30 up to line 28 so its 

```js
it('should return null users and authentication error without a token', async() => {
```
 and next `yarn` lint will scream that line length is longer than 80 😂

If anyone here knows how one is supposed to fix this in a less ugly way: pray tell.

**Comment by ojongerius** [comment]
Note to self and others, some inspiration at https://github.com/dwyl/hapi-auth-jwt2/blob/f17bac65d40b2a9154da84390b874cae4e1de192/test/test.js

And lots of interesting bits in this issue here https://github.com/dwyl/learn-json-web-tokens/issues/32 

AFAICS we get some more minimal testing in to support confidence in change for now. I'd be interesting in thinking a little more about what is good for us, discuss testing in general.

File: test/user.test.js (hunk 3 above)
Comment (Bouncey) [reply test/user.test.js:29]: `prettier` takes this out of our hands, not leading us down the path of 'how to trick prettier to write the code I like'! :grinning:  :grinning: 

You could try shortening the test message....

**Comment by Bouncey** [comment]
I am going to merge this into #58. I have been trying to get your initial tests to pass by doing some weird and wonderful things.

This looks a lot nicer/saner. :+1:

Thank you!

**Pull request closed without merging**